vsphere/stackstate_checks/vsphere/vsphere.py

```diff
@@ -509,38 +503,57 @@ def _get_server_instance(self, instance):
+                "verification. You cannot do both. Proceeding with "
+                "disabling ssl verification."
+            )
+
+        try:
+            # Object returned by SmartConnect is a ServerInstance
+            # https://www.vmware.com/support/developer/vc-sdk/visdk2xpubs/ReferenceGuide/vim.ServiceInstance.html
+            server_instance = connect.SmartConnect(
+                host=instance.get('host'),
+                user=instance.get('username'),
+                pwd=instance.get('password'),
+                sslContext=context if not ssl_verify or ssl_capath else None,
+            )
+        except Exception as e:
+            err_msg = "Connection to {} failed: {}".format(instance.get('host'), e)
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=service_check_tags, message=err_msg)
+            raise Exception(err_msg)
+
+        # Check that we have sufficient permission for the calls we need to make
+        try:
+            server_instance.CurrentTime()
+        except Exception as e:
+            err_msg = (
+                "A connection to {} can be established, but performing operations on the server fails: {}"
+            ).format(instance.get('host'), e)
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=service_check_tags, message=err_msg)
+            raise Exception(err_msg)
+
+        return server_instance
+
+    def _get_server_instance(self, instance):
+        i_key = self._instance_key(instance)
+
+        service_check_tags = [
+            'vcenter_server:{0}'.format(instance.get('name')),
+            'vcenter_host:{0}'.format(instance.get('host')),
+        ]
 
         if i_key not in self.server_instances:
-            try:
-                # Object returned by SmartConnect is a ServerInstance
-                #   https://www.vmware.com/support/developer/vc-sdk/visdk2xpubs/ReferenceGuide/vim.ServiceInstance.html
-                server_instance = connect.SmartConnect(
-                    host=instance.get('host'),
-                    user=instance.get('username'),
-                    pwd=instance.get('password'),
-                    connectionPoolTimeout=-1,
-                    sslContext=context if not ssl_verify or ssl_capath else None
-                )
-            except Exception as e:
-                err_msg = "Connection to %s failed: %s" % (instance.get('host'), e)
-                self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL,
-                                   tags=service_check_tags, message=err_msg)
-                raise Exception(err_msg)
-            self.server_instances[i_key] = server_instance
+            self.server_instances[i_key] = self._smart_connect(instance, service_check_tags)
 
         # Test if the connection is working
         try:
-            self.server_instances[i_key].RetrieveContent()
+            self.server_instances[i_key].CurrentTime()
```

Comment:
`server_instance.CurrentTime()` is already done in the `_smart_connect` function, so we don't have to do it again here

Reply:
@zandrewitte Yeah, I also thought the same but the scenario could be like if we already have `i_key` in the dictionary then we are not going into that loop. And when we do connect the first time then we also test if we have enough permission to make calls.

Reply:
Ah right of course, ok resolving this one